### DISCUSSION TITLE
List of the test examples that are expected to fail ?

### DISCUSSION THREAD

**Discussion by KipRM** [seleniumbase/SeleniumBase#2012] [Q&A] [answered]
At the end of an all example launched pytest , I noticed there are test that show failures.
Is there a list of all the tests that are expected to have failures ?
If so couldn't the test run just echo that information at the end near the test result roll-up ?

**Comment by mdmintz** [comment, marked as answer]
If running tests from https://github.com/seleniumbase/SeleniumBase/tree/master/examples with this command:

```bash
pytest --headless -n8 --dashboard --html=report.html -v --rs --crumbs
```

You can expect to see something like this with the current tests: (Runtime will be different each time, based on how fast your computer and internet connection are.)

```bash
11 failed, 123 passed, 13 skipped, 1 xfailed in 34.27s
```

Here's the `dashboard.html` file with the failures: (There's also a `report.html` file with screenshots included.)

<img width="1295" alt="Screenshot 2023-08-18 at 4 27 02 PM" src="https://github.com/seleniumbase/SeleniumBase/assets/6788579/488b4d0c-026f-4f07-ac9b-fd193bed2faa">

<div /><br />

Here's also a list of expected failures:

```
swag_labs_user_tests.py::SwagLabsTests::test_swag_labs_user_flows_1_problem_user
test_deferred_asserts.py::DeferredAssertTests::test_deferred_asserts
test_detect_404s.py::BrokenLinkTests::test_link_checking
test_fail.py::FailingTests::test_find_army_of_robots_on_xkcd_desert_island
test_suite.py::MyTestSuite::test_2
test_suite.py::MyTestSuite::test_4
test_xfail.py::XFailTests::test_xfail
time_limit_test.py::TimeLimitTests::test_runtime_limit_decorator
time_limit_test.py::TimeLimitTests::test_set_time_limit_method
visual_testing/test_layout_fail.py::VisualLayout_FixtureTests::test_python_home_change
visual_testing/test_layout_fail.py::VisualLayoutFailureTests::test_applitools_change
visual_testing/test_layout_fail.py::VisualLayoutFailureTests::test_xkcd_logo_change
```

Failing tests are included on purpose to demonstrate how SeleniumBase tests catch bugs (and log the information).
